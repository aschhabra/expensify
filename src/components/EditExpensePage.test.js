
import React from "react";
import {shallow} from "enzyme";
import {EditExpensePage} from "./EditExpensePage";
import testexpenses from "../fixtures/expenses"

let editExpense,history, wrapper,removeExpense;
beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<EditExpensePage 
                      editExpense={editExpense} 
                      removeExpense={removeExpense} 
                      history={history} 
                      expense={testexpenses[2]}
                    />);
});

test("should render edit expense",()=>{
    expect(wrapper).toMatchSnapshot();
})

test("should hanlde editExpense",()=>{

   wrapper.find("ExpenseForm").prop("onSubmit")(testexpenses[2]);
   expect(history.push).toHaveBeenLastCalledWith("/");
   expect(editExpense).toHaveBeenLastCalledWith(testexpenses[2].id,testexpenses[2]);
})
test("should handle removeExpense",()=>{

   wrapper.find("button").simulate("click");
   expect(history.push).toHaveBeenLastCalledWith("/");
   expect(removeExpense).toHaveBeenLastCalledWith({
      id:testexpenses[2].id
    });
})
