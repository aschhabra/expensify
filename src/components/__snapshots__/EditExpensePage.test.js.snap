// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit expense 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 4595,
        "createdAt": -345600000,
        "description": "Test ",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
